### DISCUSSION TITLE
como utilizar o retorne dentro de uma função

### DISCUSSION THREAD

**Discussion by Kkraties13** [dgadelha/Portugol-Webstudio#223] [Dúvidas / Perguntas] [answered]
Preciso deixar público um vetor que está em uma lista, mas quando coloco **retorne vetorx** ele simplesmente acusa erro e diz a seguinte mensagem "Não é possível retornar um valor do tipo 'inteiro' em uma função que retorna 'vazio'", como resolver isso e usar o retorne e tem como retornar mais de um tipo variável?

**Comment by dgadelha** [comment, marked as answer]
Olá!

A função que retorna um valor deve vir com o tipo de retorno na sua declaração, por exemplo:

```portugol
programa {
  funcao inteiro teste() {
    retorne 42
  }
  
  funcao inicio() {
    inteiro valor = teste()
    escreva("resposta: ", valor)
  }
}
